webui/react/src/hooks/useModal/useModalExperimentMove.tsx

```diff
@@ -1,29 +1,83 @@
-import { Empty, Select, Typography } from 'antd';
+import { Empty, notification, Select, Typography } from 'antd';
 import { ModalFuncProps } from 'antd/es/modal/Modal';
 import { SelectValue } from 'antd/lib/select';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { FixedSizeList as List } from 'react-window';
 
 import Icon from 'components/Icon';
+import Link from 'components/Link';
 import SelectFilter from 'components/SelectFilter';
-import useModal, { ModalHooks } from 'hooks/useModal/useModal';
+import useSettings, { BaseType, SettingsConfig } from 'hooks/useSettings';
+import { paths } from 'routes/utils';
 import { getWorkspaceProjects, getWorkspaces, moveExperiment } from 'services/api';
-import { ExperimentTask, Project, Workspace } from 'types';
+import { Project, Workspace } from 'types';
 import { isEqual } from 'utils/data';
 import handleError, { ErrorLevel, ErrorType } from 'utils/error';
 
+import useModal, { ModalHooks as Hooks } from './useModal';
 import css from './useModalExperimentMove.module.scss';
 
 const { Option } = Select;
 
 interface Props {
-  experiment?: ExperimentTask;
   onClose?: () => void;
 }
 
-const useModalExperimentMove = ({ onClose, experiment }: Props): ModalHooks => {
-  const [ selectedWorkspaceId, setSelectedWorkspaceId ] = useState<number>();
-  const [ destinationProjectId, setDestinationProjectId ] = useState<number>();
+export interface ShowModalProps {
+  experimentIds?: number[];
+  initialModalProps?: ModalFuncProps;
+  sourceProjectId?: number;
+  sourceWorkspaceId?: number;
+}
+
+interface ModalHooks extends Omit<Hooks, 'modalOpen'> {
+  modalOpen: (props: ShowModalProps) => void;
+}
+
+export interface Settings {
+  projectId?: number;
+  workspaceId?: number;
+}
+
+export const settingsConfig: SettingsConfig = {
+  settings: [
+    {
+      defaultValue: undefined,
+      key: 'workspaceId',
+      skipUrlEncoding: true,
+      storageKey: 'workspaceId',
+      type: { baseType: BaseType.Integer },
+    },
+    {
+      defaultValue: undefined,
+      key: 'projectId',
+      skipUrlEncoding: true,
+      storageKey: 'projectId',
+      type: { baseType: BaseType.Integer },
+    },
+  ],
+  storagePath: 'experiment-destination',
+};
+
+const moveExperimentWithHandler = async (
+  experimentId: number,
+  destinationProjectId: number,
+): Promise<number> => {
+  try {
+    await moveExperiment({ destinationProjectId, experimentId });
+    return 0;
```

Comment:
nit / question, is there a benefit to making this return 1/0 instead of true/false?

Reply:
no, i was initially going to add them but then i forgot `js` doesnt have array sum.